pkgs/development/compilers/flutter/disable-auto-update.patch

```diff
@@ -0,0 +1,95 @@
+diff --git a/bin/flutter b/bin/flutter
+index 3955f8f39..219105c1d 100755
+--- a/bin/flutter
++++ b/bin/flutter
+@@ -57,7 +57,7 @@ function retry_upgrade {
+ }
+ 
+ function upgrade_flutter () {
+-  mkdir -p "$FLUTTER_ROOT/bin/cache"
++  mkdir -p "$HOME/.cache/flutter"
+ 
+   # This function is executed with a redirect that pipes the source of
+   # this script into file descriptor 3.
+@@ -96,7 +96,7 @@ function upgrade_flutter () {
+   if hash flock 2>/dev/null; then
+     flock 3 2>/dev/null || true
+   elif hash shlock 2>/dev/null; then
+-    FLUTTER_UPGRADE_LOCK="$FLUTTER_ROOT/bin/cache/.upgrade_lock"
++    FLUTTER_UPGRADE_LOCK="$HOME/.cache/flutter/.upgrade_lock"
+     while ! shlock -f "$FLUTTER_UPGRADE_LOCK" -p $$ ; do sleep .1 ; done
+     trap _rmlock EXIT
+   fi
+@@ -185,8 +185,6 @@ fi
+ # FLUTTER_TOOL_ARGS="--enable-asserts $FLUTTER_TOOL_ARGS"
+ # FLUTTER_TOOL_ARGS="$FLUTTER_TOOL_ARGS --observe=65432"
+ 
+-(upgrade_flutter) 3< "$PROG_NAME"
+-
+ # FLUTTER_TOOL_ARGS isn't quoted below, because it is meant to be considered as
+ # separate space-separated args.
+ "$DART" --packages="$FLUTTER_TOOLS_DIR/.packages" $FLUTTER_TOOL_ARGS "$SNAPSHOT_PATH" "$@"
+diff --git a/packages/flutter_tools/lib/src/cache.dart b/packages/flutter_tools/lib/src/cache.dart
+index 5e1950b56..a0eb0a829 100644
+--- a/packages/flutter_tools/lib/src/cache.dart
++++ b/packages/flutter_tools/lib/src/cache.dart
+@@ -165,7 +165,7 @@ class Cache {
+     }
+     assert(_lock == null);
+     final File lockFile =
+-        fs.file(fs.path.join(flutterRoot, 'bin', 'cache', 'lockfile'));
++        fs.file(fs.path.join(homeDirPath, '.cache', 'lockfile'));
+     try {
+       _lock = lockFile.openSync(mode: FileMode.write);
+     } on FileSystemException catch (e) {
+@@ -356,19 +356,19 @@ class Cache {
+       if (artifact.isUpToDate()) {
+         continue;
+       }
+-      try {
+-        await artifact.update();
+-      } on SocketException catch (e) {
+-        if (_hostsBlockedInChina.contains(e.address?.host)) {
+-          printError(
+-            'Failed to retrieve Flutter tool dependencies: ${e.message}.\n'
+-            'If you\'re in China, please see this page: '
+-            'https://flutter.dev/community/china',
+-            emphasis: true,
+-          );
+-        }
+-        rethrow;
+-      }
++      // try {
++      //   await artifact.update();
++      // } on SocketException catch (e) {
++      //   if (_hostsBlockedInChina.contains(e.address?.host)) {
++      //     printError(
++      //       'Failed to retrieve Flutter tool dependencies: ${e.message}.\n'
++      //       'If you\'re in China, please see this page: '
++      //       'https://flutter.dev/community/china',
++      //       emphasis: true,
++      //     );
++      //   }
++      //   rethrow;
++      // }
```

Comment:
This is removed because we have an issue with missing dependency.
To fix this, I need to simulate the precache command and download all dependencies by "hand".

Reply:
Solution: cache is now in $HOME/.cache/flutter